File: utilities/conf.cpp
#include <iostream>
#include <GraphMol/MolOps.h>
#include <GraphMol/FileParsers/MolSupplier.h>
#include <GraphMol/FileParsers/MolWriters.h>
#include <GraphMol/Descriptors/Lipinski.h>
#include <GraphMol/DistGeomHelpers/Embedder.h>
#include <GraphMol/ForceFieldHelpers/UFF/Builder.h>
#include <ForceField/ForceField.h>
using namespace std;
using namespace RDKit;
using namespace RDKit::MolOps;
using namespace RDKit::Descriptors;
using namespace RDKit::DGeomHelpers;
using namespace RDKit::UFF;

int main()
{
	SDWriter writer(&cout);
	for (SmilesMolSupplier sup(&cin, false, "\t", 1, 0); !sup.atEnd();)
	{
		const auto mol = addHs(*sup.next());
		const auto nrot = calcNumRotatableBonds(*mol);
		const auto numConfs = nrot <= 7 ? 50 : (nrot <= 12 ? 200 : 300);
		for (const auto id : EmbedMultipleConfs(*mol, numConfs))
		{
			auto ff = constructForceField(*mol, 100.0, id);
			ff->initialize();
			ff->minimize();
			const auto e = ff->calcEnergy();
//			writer.write(*mol, id);
		}
	}
}

File: utilities/conf.py
#!/usr/bin/env python2
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdMolDescriptors

writer = Chem.SDWriter("16.sdf")
sup = Chem.SmilesMolSupplier("16.smi", delimiter='\t', smilesColumn=1, nameColumn=0)
for mol in sup:
	mol = Chem.AddHs(mol)
	nrot = rdMolDescriptors.CalcNumRotatableBonds(mol)
	numConfs = 50 if nrot <= 7 else 200 if nrot <= 12 else 300
	numConfs = 2
	for id in AllChem.EmbedMultipleConfs(mol, numConfs):
#		AllChem.UFFOptimizeMolecule(mol, confId=id)
		ff = AllChem.UFFGetMoleculeForceField(mol, confId=id)
		ff.Initialize()
		ff.Minimize()
		e = ff.CalcEnergy()
		rmsd = AllChem.GetBestRMS(mol, mol, 0, id)
		writer.write(mol, confId=id)
